plugin/network_processor.go

```diff
@@ -0,0 +1,259 @@
+package plugin
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/compute/mgmt/compute"
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/network/mgmt/network"
+)
+
+// -----------------------------------------------------------------------------
+// Types
+// -----------------------------------------------------------------------------
+type networkInfo struct {
+	IpAddresses []string
+}
+
+// Common interface for IP configuration processing
+type networkProcessor interface {
+	processIPConfigurations(ctx context.Context, ipConfigs *[]network.InterfaceIPConfiguration) error
```

Comment:
Ditto no pointer to slice
```suggestion
	processIPConfigurations(ctx context.Context, ipConfigs []network.InterfaceIPConfiguration) error
```

Reply:
I think I would also prefer if this returned the netInfo instead of writing it to the processor. I think it makes the data flow easier to read
```suggestion
	processIPConfigurations(ctx context.Context, ipConfigs *[]network.InterfaceIPConfiguration) (*networkInfo. error)
```